src/plugins/vis_types/timeseries/public/convert_to_lens/lib/xy/date_histogram.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DataViewsPublicPluginStart } from '@kbn/data-views-plugin/public';
+import { Series } from '../../../../common/types';
+import { getFieldType } from '../datasource';
+
+export const isSplitWithDateHistogram = async (
+  series: Series,
+  splitFields: string[],
+  indexPatternId: string,
+  dataViews: DataViewsPublicPluginStart
+) => {
+  let splitWithDateHistogram = false;
+  if (series.terms_field && series.split_mode === 'terms' && splitFields) {
+    for (const f of splitFields) {
+      const fieldType = await getFieldType(indexPatternId, f, dataViews);
+
+      if (fieldType === 'date') {
+        if (splitFields.length === 1) {
+          splitWithDateHistogram = true;
```

Comment:
Can we return `true` here instead of keep looping?

Reply:
Unfortunately no, because we should check all splitFields because if splitFields.length !== 1 we should return null instead of true

Reply:
I think I'm lost: `splitFields` will be 1 for all iterations here, no? Or did you mean to check for `f.length` instead?

Reply:
Honestly as I see we moved logic from file which created by @stratoula. As I understand we don't support if we split by date and something else. Because of that we check firstly "fieldType === 'date'" and after that splitFields.length